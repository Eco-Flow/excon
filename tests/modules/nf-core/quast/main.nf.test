nextflow_process {

    name "Test Process QUAST"
    script "modules/nf-core/quast/main.nf"
    process "QUAST"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                clean = false
                outdir = "results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of( ["Drosophila_santomea", file("${projectDir}/data/gffread_input/genome.fna.gz")] )
                input[1] = Channel.of( ["Drosophila_santomea", file("${projectDir}/data/gffread_input/genomic.gff.gz")] )
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                assert results.size() == 1
                // assert results[0][1].size() == 12
            }
            //Assert snapshot of .tsv file with summary
            assert snapshot(
                path("${launchDir}/${params.outdir}/quast/Drosophila_santomea.tsv"),
            ).match("quast_stats")
        }

    }

}
