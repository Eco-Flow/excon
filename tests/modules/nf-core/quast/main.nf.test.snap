{
    "quast_stats": {
        "content": [
            "Drosophila_santomea.tsv:md5,92a5bc9851fdc82fc66bbcaae17557f7"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-20T15:09:08.942996008"
    }
}